### ISSUE TITLE
Validation Plan: Test strategy effectiveness without backtest bias

### ISSUE THREAD

**Issue by kumargu** [kumargu/OpenQuant#95]
## Context (Updated post-sprint)

Sprint delivered 8 PRs: P&L went from **-$1,495 to +$963** with max drawdown reduced 19% ($1,134 → $919). The improvements came from:

1. **GJR-GARCH** — forward-looking asymmetric vol (PR #91)
2. **BOCPD regime detection** — 4 regimes: LowVol/Normal/HighVol/Crisis (PR #92)
3. **Regime-conditional combiner** — suppress conflicting strategies per regime (PR #93)
4. **Bayesian Kelly** — uncertainty-penalized position sizing (PR #94)
5. **Drawdown deleveraging** — smooth `(1 - dd_ratio)²` curve replacing binary kill switch (PR #96)
6. **Purged CV + PBO** — validation tooling now available (PR #97)
7. **Meta-labeling** — offline classifier for signal quality (PR #98)

### Free Parameter Count

| Source | Parameters | Risk |
|--------|-----------|------|
| BOCPD hazard | 1 (250) | Sensitive — controls regime change speed |
| Vol percentile thresholds | 2 (0.30, 0.70) | Moderate — defines regime boundaries |
| Regime weights | 16 (4 strategies × 4 regimes) | High — most degrees of freedom |
| Kelly regime fractions | 4 (0.6, 0.5, 0.25, 0.10) | Low — follow from half/quarter-Kelly convention |
| DD deleverage curve | 1 (exponent=2) | Low — concave is standard |
| GJR-GARCH | 4 (ω, α, β, γ) | Low — well-established defaults from literature |
| **Total** | **~28** | |

With ~28 parameters and ~600 trades in backtest, we have ~21 trades per parameter. Rule of thumb wants 10x (280+), so we're at **2x the minimum**. Not great, not terrible. Validation is essential.

### The +$963 is More Convincing Than +$102

The earlier +$102 was borderline. At +$963 on 624 trades with profit factor 1.01, the signal is weak but present. The drawdown reduction (19%) is actually the more reliable signal — it doesn't depend on parameter fine-tuning, it comes from the structural change (drawdown deleveraging curve).

---

#### 1. Run Purged CV — Already Built (PR #97) ⏱️ 30 min

```bash
# Run CPCV on your main symbols
python -m paper_trading.purged_cv --symbol BTC/USD --days 30 --n-groups 6
python -m paper_trading.purged_cv --symbol AAPL --days 30 --n-groups 6
```

**Interpret:**
- PBO < 0.30 → **GO** — low probability of overfitting
- PBO 0.30-0.50 → **CAUTION** — paper trade with close monitoring
- PBO > 0.50 → **NO GO** — parameters are likely overfit

#### 2. Parameter Sensitivity Sweep ⏱️ 1 hour

The most important test. Vary each critical parameter and check if P&L stays positive:

```python
# BOCPD hazard: test 2x range around 250
for hazard in [125, 150, 200, 250, 300, 350, 400, 500]:
    pnl = run_backtest(bocpd_hazard=hazard)
    print(f"hazard={hazard}: PnL=${pnl:.0f}")
    
# Vol thresholds: test grid
for lo in [0.20, 0.25, 0.30, 0.35, 0.40]:
    for hi in [0.60, 0.65, 0.70, 0.75, 0.80]:
        pnl = run_backtest(vol_low=lo, vol_high=hi)
        
# Regime weights: test equal weights within regime (simplification test)
# If equal regime weights still improve over no regime detection,
# the REGIME DETECTION is adding value (not the specific weights)
pnl_equal = run_backtest(regime_weights="equal")
pnl_no_regime = run_backtest(regime_detection=False)
```

**PASS if:** 
- Strategy profitable across ≥ 2x range of hazard (125-500)
- Strategy profitable across ≥ 3 threshold combinations
- Equal regime weights still beat no-regime baseline

**FAIL if:** Only hazard=250 with thresholds=(0.30, 0.70) works.

#### 3. Regime Permutation Test ⏱️ 2 hours

The definitive test — are the regime labels adding real signal?

```python
import numpy as np

# Collect regime labels from a backtest run
actual_sharpe = compute_sharpe(strategy_with_real_regimes)

# Permutation test: shuffle regime labels 1000x
random_sharpes = []
for _ in range(1000):
    shuffled_regimes = np.random.permutation(regime_labels)
    sr = compute_sharpe(strategy_with_shuffled_regimes)
    random_sharpes.append(sr)

p_value = np.mean(np.array(random_sharpes) >= actual_sharpe)
print(f"Regime detection p-value: {p_value:.4f}")
```

**PASS if:** p-value < 0.05 (fewer than 5% of random regimes beat real ones).
**FAIL if:** p-value > 0.05 (regime detection isn't adding real signal).

#### 4. Run Meta-Labeling — Already Built (PR #98) ⏱️ 30 min

```bash
python -m paper_trading.meta_labeling --category crypto --days 30
```

**Check:**
- AUC > 0.55 — meta-model can distinguish profitable from unprofitable signals
- Feature importances — if `garch_vol` and `regime_change_prob` rank high, the new features are adding value
- If AUC ≈ 0.50 — meta-model can't tell, signals may be noise

#### 5. Walk-Forward Analysis
Rolling 2-year IS / 6-month OOS (or scaled for your data: 20-day IS / 5-day OOS for intraday):
- WFE (Walk-Forward Efficiency) > 0.50 per window
- No more than 1 in 4 windows negative
- No structural decline in OOS Sharpe

#### 6. Bootstrap Sharpe CI
```python
import numpy as np

def bootstrap_sharpe(returns, n_boot=10000):
    sharpes = []
    for _ in range(n_boot):
        sample = np.random.choice(returns, size=len(returns), replace=True)
        sr = np.sqrt(252) * sample.mean() / max(sample.std(), 1e-10)
        sharpes.append(sr)
    return np.percentile(sharpes, 2.5), np.percentile(sharpes, 97.5)

lo, hi = bootstrap_sharpe(daily_returns)
print(f"Sharpe 95% CI: [{lo:.2f}, {hi:.2f}]")
```

**PASS if:** Lower bound > 0.
**FAIL if:** CI includes 0.

#### 7. Deflated Sharpe Ratio

```python
# Be honest about N — count ALL configurations you tried
N_trials = 28  # at minimum: all free parameters
```

**PASS if:** DSR p-value < 0.05.

#### 8. Paper Trade (3-6 months)
- Minimum 100 trades
- Must see strategy perform in ≥ 2 distinct regimes
- Pre-register benchmark: **"Sharpe > 0.3 over 100+ trades"**
- **DO NOT touch parameters during this period**
- Track regime transitions — does the engine correctly identify them?

#### 9. Gradual Capital Allocation
- Start at 10% of target allocation
- Scale up over 3 months if live performance within tolerance
- Auto-kill: halt if drawdown exceeds 1.5× backtest max DD ($919 × 1.5 = $1,379)

---

## Minimum Bar for Deployment

| Metric | Minimum | Current | How to Get |
|--------|---------|---------|-----------|
| PBO | < 0.40 | **Unknown** | `python -m paper_trading.purged_cv` |
| Param sensitivity | Profitable across 2x range | **Unknown** | Sweep script |
| Regime permutation p-value | < 0.05 | **Unknown** | Permutation test |
| Meta-label AUC | > 0.55 | **Unknown** | `python -m paper_trading.meta_labeling` |
| Bootstrap Sharpe CI lower | > 0 | **Unknown** | Bootstrap script |
| DSR p-value | < 0.05 | **Unknown** | DSR computation |
| Paper trade period | 3+ months | 0 | Start today |
| Minimum trades (paper) | 100+ | 0 | Accumulate |

---

## Decision Framework for Today

```
Run PBO (#1)
  ├── PBO < 0.30 → Run param sensitivity (#2)
  │     ├── Profitable across 2x range → PAPER TRADE TODAY ✅
  │     └── Only works at exact params → DO NOT TRADE ❌
  ├── PBO 0.30-0.50 → Run param sensitivity + permutation test (#2, #3)
  │     ├── Both pass → PAPER TRADE with tight monitoring ⚠️
  │     └── Either fails → DO NOT TRADE ❌
  └── PBO > 0.50 → DO NOT TRADE ❌
```

## What We're Really Testing

The structural improvements (GARCH, regime detection, Kelly sizing, drawdown deleveraging) are **architecturally sound** — they're grounded in well-established quant finance literature. The question isn't "is regime detection a good idea?" (it is). The question is: **"are these specific parameter values (hazard=250, thresholds=0.30/0.70) robust, or were they fit to noise?"**

If the parameter sensitivity test passes (profitable across a wide range), then the exact values don't matter much and the approach is sound. If it fails, you need to simplify — perhaps use fixed tercile thresholds (33rd/67th percentile) that require zero parameter choice.

## Python Libraries

```
pip install pypbo pybroker vectorbt arch scipy scikit-learn
```

## References
- Bailey et al. (2017) "The Probability of Backtest Overfitting" — [SSRN](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=2326253)
- Bailey & Lopez de Prado "The Deflated Sharpe Ratio" — [PDF](https://www.davidhbailey.com/dhbpapers/deflated-sharpe.pdf)
- [pypbo implementation](https://github.com/esvhd/pypbo)
- [Walk-Forward Analysis Guide](https://www.interactivebrokers.com/campus/ibkr-quant-news/the-future-of-backtesting-a-deep-dive-into-walk-forward-analysis/)
- Grossman & Zhou (1993) "Optimal Investment Strategies for Controlling Drawdowns"
- De Prado (2018) "Advances in Financial Machine Learning" Ch. 7, 12
